src/dist/replication/meta_server/meta_bulk_load_service.cpp

```diff
@@ -29,6 +30,232 @@ void bulk_load_service::initialize_bulk_load_service()
     }
 }
 
+// ThreadPool: THREAD_POOL_META_STATE
+void bulk_load_service::on_start_bulk_load(start_bulk_load_rpc rpc)
+{
+    const auto &request = rpc.request();
+    auto &response = rpc.response();
+    response.err = ERR_OK;
+
+    std::shared_ptr<app_state> app;
+    {
+        zauto_read_lock l(app_lock());
+
+        app = _state->get_app(request.app_name);
+        if (app == nullptr || app->status != app_status::AS_AVAILABLE) {
+            derror_f("app({}) is not existed or not available", request.app_name);
+            response.err = app == nullptr ? ERR_APP_NOT_EXIST : ERR_APP_DROPPED;
+            response.hint_msg = fmt::format(
+                "app {}", response.err == ERR_APP_NOT_EXIST ? "not existed" : "dropped");
+            return;
+        }
+
+        if (app->is_bulk_loading) {
+            derror_f("app({}) is already executing bulk load, please wait", app->app_name);
+            response.err = ERR_BUSY;
```

Comment:
I think `ERR_SERVICE_ALREADY_RUNNING` is better here.

Reply:
I think ERR_BUSY is ok, the error_code and hint_message is enough for users to know why starting bulk load is failed. Besides, in my view, service is like network, bulk load may be not qualified to be a service.